pep-0484.txt

```diff
@@ -984,15 +984,16 @@ for example, the above is equivalent to::
 
   def handle_employee(e: Optional[Employee]) -> None: ...
 
-An optional type is also automatically assumed when the default value is
-``None``, for example::
+Type checkers may also automatically assume an optional type when the
+default value is ``None``, for example::
 
   def handle_employee(e: Employee = None): ...
 
-This is equivalent to::
+This may be treated as equivalent to::
 
   def handle_employee(e: Optional[Employee] = None) -> None: ...
 
+Type checkers may also require the optional type to be made explicit.
```

Comment:
Hm, maybe I was too wishy-washy. Pondering this again I think the language in PEP 484 should acknowledge that PEP 484 used to automatically infer `Optional[]` based on an `= None` default, and that type-checkers historically did it that way, but that we changed our mind and that in the future the PEP will mandate that this *not* be done. What type checkers *currently* do is then merely dependent on where the type checker is in its migration to the future behavior.

Does that make sense?

Reply:
@JelleZijlstra Any response to that?

Reply:
I forgot about this one, will get back to it soon.

Reply:
How is this new wording?